qa/tasks/cephfs/test_uninlining.py

```diff
@@ -0,0 +1,323 @@
+        self.snapshot_info[1] = "with snapshots"
+
+    def tearDown(self):
+        super(TestDataUninlining, self).tearDown()
+
+    def extract_inodes(self, files):
+        inodes = []
+        for fil in files:
+            log.debug(f"getting inode for:{fil}")
+            cmd = f'ls -i {fil}'
+            o = remote_mntpt_cmd(self.mount_a, cmd)
+            inodes.append(o.split(' ')[0])
+        return inodes
+
+    def get_inline_data_info(self, inodes, files, dir_pins, num_mds):
+        def get_inode_dump(inode, rank, retries):
+            for i in range(retries):
+                log.debug(f"try #{i+1} - dump inode {inode}")
+                try:
+                    json_out = self.fs.rank_tell(['dump', 'inode', inode], rank)
+                    return json_out
+                except json.decoder.JSONDecodeError:
+                    time.sleep(1)
+            raise json.decoder.JSONDecodeError(f'No JSON found after {retries} attempts', None, 0)
+
+        info = []
+        for i in range(len(inodes)):
+            inode = inodes[i]
+            log.debug(f"getting inode info #{i+1} of {len(inodes)}:{inode}")
+            path = os.path.dirname(files[i])
+            rank = dir_pins[path] if path in dir_pins else 0
+            r = rank
+            while r < rank + num_mds:
+                try:
+                    json_out = get_inode_dump(inode,
+                                              r % num_mds,
+                                              self.DUMP_INODE_RETRIES)
+                    break
+                except json.decoder.JSONDecodeError:
+                    pass
+                finally:
+                    r += 1
+            self.assertTrue(json_out is not None)
+            self.assertTrue('inline_data_length' in json_out)
+            self.assertTrue('inline_data_version' in json_out)
+            info.append(InlineDataInfo(json_out['inline_data_length'],
+                                       json_out['inline_data_version']))
+        return info
+
+    def run_test_worker(self,
+                        opt_clear_cache,
+                        opt_unmount,
+                        opt_mount_openbg,
+                        opt_multimds,
+                        opt_snapshot):
+        log.info("Running Data Uninlining test with: "
+                 f"{self.cache_info[opt_clear_cache]}, "
+                 f"{self.unmount_info[opt_unmount]}, "
+                 f"{self.mount_openbg_info[opt_mount_openbg]}, "
+                 f"{self.multimds_info[opt_multimds]}, "
+                 f"{self.snapshot_info[opt_snapshot]}")
+
+        # Set max_mds to 1 or 2
+        num_mds = 2 if opt_multimds else 1
+        log.debug(f"setting max_mds:{num_mds}")
+        self.fs.set_max_mds(num_mds)
+
+        # Get configured max inline data size
+        log.debug("getting client_max_inline_size")
+        idsize = self.fs.fs_config.get('client_max_inline_size', 4096)
```

Comment:
IMO we should strict the mounts to `ceph-fuse`. Because the kernel mount is buggy when the `client_max_inline_size` > 4KB. 

Or you could just force set `client_max_inline_size` to 4KB for this test ? Which could work for both kclient and fuse mounts.

Reply:
It's okay, just want to make sure in future someone won't touch this in this test.

And maybe we can just add some comments to give the users a hint if they hit issues between fuse and kernel clients for their existing inline data usage.

Reply:
I'll add a comment in the python test code.